### PR TITLE
stb_vorbis bug fix: audiofiles with empty comment section was unable to load

### PR DESCRIPTION
**Pull request by romanshuvalov**
Fixed an error which triggered VORBIS_outofmem when loading a file without comment section (comment_list_length==0). Also fixed couple of warnings about unused parameters.

### CODE REVIEW

File: stb_vorbis.c [hunk 1]
```diff
@@ -3642,18 +3647,23 @@ static int start_decoder(vorb *f)
...
-         f->comment_list[i][j] = get8_packet(f);
-      }
-      f->comment_list[i][len] = (char)'\0';
+          for(j=0; j < len; ++j) {
+             f->comment_list[i][j] = get8_packet(f);
+          }
+          f->comment_list[i][len] = (char)'\0';
+       }
+   }
+   else {
```
Comment (fluffrabbit) [inline stb_vorbis.c:3665]: `else` should be on the same line as the closing bracket.

File: stb_vorbis.c [hunk 2]
```diff
@@ -4191,10 +4201,12 @@ static void vorbis_deinit(stb_vorbis *p)
...
-   for (i=0; i < p->comment_list_length; ++i) {
-      setup_free(p, p->comment_list[i]);
-   }
-   setup_free(p, p->comment_list);
+   if (p->comment_list != NULL) {
+       for (i=0; i < p->comment_list_length; ++i) {
+          setup_free(p, p->comment_list[i]);
+       }
+       setup_free(p, p->comment_list);
+   };
```
Comment (fluffrabbit) [inline stb_vorbis.c:4209]: Unnecessary semicolon?

File: stb_vorbis.c [hunk 3]
```diff
@@ -904,6 +905,7 @@ static int error(vorb *f, enum STBVorbisError e)
    return 0;
 }
 
+#define UNUSED_PARAMETER(x) (void)(x)
```
Comment (fluffrabbit) [inline stb_vorbis.c:908]: This is a matter of opinion, but is defining `UNUSED_PARAMETER` necessary, as most people know what `(void)foo` does? I could see the use case of searching for the macro to find dead parameters, but only in cases where it's an actual dead parameter and not something that is compiled conditionally.

**Review by fluffrabbit — COMMENTED** [review]
The latest revision of mainline stb_vorbis broke files without comments. Something to do with it trying to throw errors and whatnot. I want this bug fixed, but I have a couple of comments on this PR.

I'll probably use this patch in production like now, so thank you in any case. :)

**Comment by ROTARTSI82** [comment]
I noticed this bug in my code when it suddenly stopped working :\

Also, this can be fixed by only changing this to ` if (f->comment_list_length && f->comment_list == NULL) ...`
https://github.com/nothings/stb/blob/c9064e317699d2e495f36ba4f9ac037e88ee371a/stb_vorbis.c#L3646
and adding that null check. Why is free called in a loop instead of just one free call to `comment_list`?

**Comment by nothings** [comment]
Integrated a different PR with this fix. Thank you for reporting, my apologies for how long it took to address.

**Pull request closed without merging**